packages/utils/src/format.ts

```diff
@@ -27,3 +28,29 @@ export function truncBytes(root: Uint8Array | string): string {
   const str = typeof root === "string" ? root : toHexString(root);
   return str.slice(0, 14);
 }
+
+/**
+ * Format a bigint value as a decimal string
+ */
+export function formatBigDecimal(numerator: bigint, denominator: bigint, maxDecimalFactor: bigint): string {
+  const full = numerator / denominator;
+  const fraction = ((numerator - full * denominator) * maxDecimalFactor) / denominator;
+
+  // zeros to be added post decimal are number of zeros in maxDecimalFactor - number of digits in fraction
+  const zerosPostDecimal = String(maxDecimalFactor).length - 1 - String(fraction).length;
+  return `${full}.${"0".repeat(zerosPostDecimal)}${fraction}`;
+}
+
+// display upto 5 decimal places
+const MAX_DECIMAL_FACTOR = BigInt("100000");
+
+/**
+ * Format wei as ETH, with up to 5 decimals
+ *
+ * if suffix is true, append ' ETH'
+ */
+export function prettyWeiToEth(wei: bigint, suffix = false): string {
```

Comment:
curious, is there a case where we want to call this an not append ETH suffix or should we consider that part of why it's "pretty" and not just decimals?

(assumption here is that this function is used solely for logging purposes as most of the other functions in this file)

Reply:
I am not sure, I believe @wemeetagain added it for future use cases. e.g. we log and the key names already in form `totalBlockValueEth` then we don't need to suffix the eth for logging.

Reply:
> we log and the key names already in form totalBlockValueEth

I don't think we do, to support better readability I think we should prevent to add suffixes to arleady long variable names.

For consistency with VC logs I think having ` ETH` suffix is better, we had a discussion on that PR already about readability of different approaches and went with that format.